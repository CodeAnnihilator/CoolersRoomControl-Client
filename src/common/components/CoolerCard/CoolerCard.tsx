import React, {Component} from 'react';
import {View, Text, ScrollView} from 'react-native';

import {ICooler} from '../../../screens/Rooms/types/Cooler';

import styles from './coolerCardStyles';

interface IProps {
	cooler: ICooler;
}

export default class CoolerCard extends Component<IProps> {
	public render() {
		const {
			cooler: {
				title, series, roomID, place, source, type, certification, cop, power,
			},
		} = this.props;
		const coolerOptions = [
			{
				title: 'Place of Origin',
				description: place,
			},
			{
				title: 'Power Source',
				description: source,
			},
			{
				title: 'Type',
				description: type,
			},
			{
				title: 'Certification',
				description: certification,
			},
			{
				title: 'COP',
				description: cop,
			},
			{
				title: 'Power (W)',
				description: power,
			},
		];

		return (
			<ScrollView>
				<View style={styles.titleWrapper}>
					<Text style={styles.title}>{title}</Text>
					<Text style={styles.series}>{series}</Text>
					<Text style={styles.room}>Room: {roomID !== null ? roomID : 'No Room'}</Text>
				</View>
				{
					coolerOptions.map(option => (
						<View style={styles.optionsWrapper} key={option.title}>
							<Text style={styles.option}>{option.title}</Text>
							<Text style={styles.description}>{option.description}</Text>
						</View>
				))}
			</ScrollView>
		);
	}
}
