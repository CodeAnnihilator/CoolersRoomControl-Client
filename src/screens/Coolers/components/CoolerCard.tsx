import React, {Component} from 'react';
import {SafeAreaView, Text} from 'react-native';

export default class CoolerCard extends Component {
	render() {
		const {navigation} = this.props;
	
		return (
			<SafeAreaView>
				<Text>
					{JSON.stringify(navigation.getParam('id', 'NO-ID'))}
				</Text>
			</SafeAreaView>
		)
	}
}
