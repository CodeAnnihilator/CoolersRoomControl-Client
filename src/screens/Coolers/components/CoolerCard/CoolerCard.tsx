import React, {Component} from 'react';
import {View, Text, ScrollView} from 'react-native';

import styles from './coolerCardStyles';

interface IProps {
	cooler: {
		id: number;
		power: string;
		series: string;
		status: string;
		title: string;
		room: string;
		place: string;
		source: string;
		type: string;
		certification: string;
		cop: string;
	};
}

export default class CoolerCard extends Component<IProps> {
	public render() {
		const {cooler} = this.props;

		return (
			<ScrollView>
				<View style={styles.titleWrapper}>
					<Text style={styles.title}>{cooler.title}</Text>
					<Text style={styles.series}>{cooler.series}</Text>
					<Text style={styles.room}>Room: {cooler.room}</Text>
				</View>
				<View style={styles.optionsWrapper}>
					<Text style={styles.option}>Place of Origin</Text>
					<Text style={styles.description}>{cooler.place}</Text>
				</View>
				<View style={styles.optionsWrapper}>
					<Text style={styles.option}>Power Source</Text>
					<Text style={styles.description}>{cooler.source}</Text>
				</View>
				<View style={styles.optionsWrapper}>
					<Text style={styles.option}>Type</Text>
					<Text style={styles.description}>{cooler.type}</Text>
				</View>
				<View style={styles.optionsWrapper}>
					<Text style={styles.option}>Certification</Text>
					<Text style={styles.description}>{cooler.certification}</Text>
				</View>
				<View style={styles.optionsWrapper}>
					<Text style={styles.option}>COP</Text>
					<Text style={styles.description}>{cooler.cop}</Text>
				</View>
				<View style={styles.optionsWrapper}>
					<Text style={styles.option}>Power (W)</Text>
					<Text style={styles.description}>{cooler.power}</Text>
				</View>
			</ScrollView>
		);
	}
}
