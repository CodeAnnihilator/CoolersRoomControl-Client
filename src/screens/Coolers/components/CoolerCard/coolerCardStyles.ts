import EStyleSheet from 'react-native-extended-stylesheet';

export default EStyleSheet.create({
	titleWrapper: {
		flex: 1,
		alignItems: 'center',
		padding: 25,
		borderBottomWidth: 2,
		borderColor: '$gray100',
	},
	optionsWrapper: {
		paddingVertical: 15,
		paddingHorizontal: 25,
		borderBottomWidth: 2,
		borderColor: '$gray100',
	},
	title: {
		fontSize: 17,
		fontWeight: 'bold',
		marginBottom: 7,
	},
	series: {
		fontSize: 11,
		fontWeight: 'bold',
		marginBottom: 14,
	},
	room: {
		fontSize: 13,
		fontWeight: 'bold',
	},
	option: {
		fontSize: 10,
		color: '$gray600',
		marginBottom: 7,
	},
	description: {
		fontSize: 12,
		fontWeight: 'bold',
		color: '$black',
	},
});
