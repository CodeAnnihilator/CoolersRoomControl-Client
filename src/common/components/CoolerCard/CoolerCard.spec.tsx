import * as React from 'react';
import renderer from 'react-test-renderer';

import CoolerCard from './CoolerCard';

describe('CoolerCard', () => {
	it('+++capturing Snapshot of CoolerCard', () => {
		const cooler = {
			id: 1,
			power: 'power',
			title: 'title',
			series: 'series',
			status: 'status',
			roomID: 0,
			place: 'place',
			source: 'source',
			type: 'type',
			certification: 'certification',
			cop: 'cop',
		};
		const renderedValue = renderer.create(<CoolerCard cooler={cooler} />).toJSON();

		expect(renderedValue).toMatchSnapshot();
	});
});
