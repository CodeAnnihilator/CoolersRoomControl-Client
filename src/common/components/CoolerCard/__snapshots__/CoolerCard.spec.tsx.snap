// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoolerCard +++capturing Snapshot of CoolerCard 1`] = `null`;
